V3WebSocketProtocolGuide.md

```diff
@@ -0,0 +1,326 @@
+
+Additionally, the WebSocket framing decided by one tunnel peer is not guaranteed to be the same as those received by the other side. For example, the maximum WebSocket frame size in the `aws.iot.securetunneling-3.0` protocol is 131076 bytes, and the service may aggregate data to a point that aggregates multiple messages to this size into a single frame. The tunneling message frames generated by a tunnel peer are maintained by the service and cannot be aggregated or fragmented. This enables known tunnel peers to operate under more restrictive guidelines than what is valid in this protocol guide. One example of this is reducing the maximum payload of a tunneling message to 16kb down from 64kb to enable local proxy implementations to reduce the size of processing buffers.
+
+### ProtocolBuffers Message Schema
+
+The data that must be parsed into a ProtocolBuffers message conforms to the following schema:
+
+```
+syntax = "proto3";
+
+package com.amazonaws.iot.securedtunneling;
+
+option java_outer_classname = "Protobuf";
+option optimize_for = LITE_RUNTIME;
+
+message Message {
+    Type    type         = 1;
+    int32   streamId     = 2;
+    bool    ignorable    = 3;
+    bytes   payload      = 4;
+    string  serviceId    = 5;
+    repeated string availableServiceIds = 6;
+    uint32 connectionId = 7;
+    
+    enum Type {
+        UNKNOWN = 0;
+        DATA = 1;
+        STREAM_START = 2;
+        STREAM_RESET = 3;
+        SESSION_RESET = 4;
+        SERVICE_IDS = 5;
+        CONNECTION_START = 6;
+        CONNECTION_RESET = 7;
+    }
+}
+```
+
+Tunneling frames (without the data length prefix) must parse into a _Message_ object and satisfy the following rules:
+
+- _Type_ field must be set to a non-zero enum value. Due to ProtocolBuffers schema recommendation, the keyword 'required' is not used in the actual schema
+- It is invalid for a client connected with mode=destination to send a message with _Type_ = _StreamStart_ over the tunnel.
+- It is invalid for any client to send message types associated with a stream (_StreamStart_, _ConnectionStart_, _Data_, _StreamReset_, _ConnectionReset_) with a stream ID of 0
+- It is invalid for a v3 client to send a message type associated with a connection (_StreamStart_, _ConnectionStart_, _Data_, _ConnectionReset_) with a connection ID of 0 will be mapped to connection ID = 1.
```

Comment:
The word `invalid` is confusing me. Maybe just simpler to say that connection ID of `0` will always be interpreted as connection ID = `1`.